src/en/mangaowl/src/eu/kanade/tachiyomi/extension/en/mangaowl/MangaOwl.kt

```diff
@@ -0,0 +1,251 @@
+                    }
+                    is StatusFilter -> if (!filter.toUriPart().isNullOrEmpty()) {
+                        url.addQueryParameter("status", filter.toUriPart())
+                    }
+                    is GenreFilter ->
+                        filter.state
+                            .filter { it.state }
+                            .forEach { url.addQueryParameter("genres", it.uriPart) }
+                    else -> {}
+                }
+            }
+
+            url.apply {
+                addQueryParameter("page", page.toString())
+            }
+            GET(url.toString(), headers)
+        }
+    }
+
+    override fun searchMangaSelector() = popularMangaSelector()
+
+    override fun searchMangaFromElement(element: Element): SManga = popularMangaFromElement(element)
+
+    override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
+
+    // Manga summary page
+
+    override fun mangaDetailsParse(document: Document): SManga {
+        val infoElement = document.select("#main-container > div > div.row-responsive > div.flex-fill.column > div > div.flex-fill.column.section-container.p-3 > div.row-responsive.py-2")
+
+        return SManga.create().apply {
+            title = infoElement.select("div.comic-info-container > h1.story-name").text()
+            infoElement.select("div.comic-info-container > div.comic-attrs > div").forEach { attr ->
+                when (attr.select("span").text()) {
+                    "Author" -> author = attr.select("div > a").text()
+                    "Genres" -> genre = attr.select("div > a").text()
+                }
+            }
+
+            description = infoElement.select("div.comic-info-container > span.story-desc").text()
+            thumbnail_url = infoElement.select("img").attr("abs:src")
+            infoElement.select("div:first-child > div.section-status").forEach { stt ->
+                if (stt.select("span:first-child").text() == "Status") {
+                    status = stt.select("span:last-child").text().let {
+                        when {
+                            it.contains("Ongoing") -> SManga.ONGOING
+                            it.contains("Completed") -> SManga.COMPLETED
+                            else -> SManga.UNKNOWN
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    // Chapters
+
+    // Only selects chapter elements with links, since sometimes chapter lists have unlinked chapters
+    override fun chapterListSelector() = "div.chapters-container > a"
+
+    private fun DateFormat.tryParse(str: String?): Long = if (str.isNullOrEmpty()) {
+        assert(false) { "Date string is null or empty" }
+        0L
+    } else {
+        runCatching {
+            parse(str)?.time ?: 0L
+        }.onFailure {
+            assert(false) { "Cannot parse date $str: ${it.message}" }
+        }.getOrDefault(0L)
+    }
```

Comment:
```suggestion
    private fun DateFormat.tryParse(str: String): Long = try {
        parse(str)!!.time
    } catch (_: ParseException) {
        0L
    }
```

Use normal try/catch. `runCatching` has the overhead of boxing the result into a `Result`. Also remove debugging assertions.